querydsl-sql/src/main/java/com/querydsl/sql/Configuration.java

```diff
@@ -223,7 +223,7 @@ public String getColumnOverride(SchemaAndTable key, String column) {
     @SuppressWarnings({ "unchecked", "rawtypes" })
     public <T> void set(PreparedStatement stmt, Path<?> path, int i, T value) throws SQLException {
         if (Null.class.isInstance(value)) {
-            Integer sqlType = path != null ? ColumnMetadata.getColumnMetadata(path).getJdbcType() : null;
+            Integer sqlType = path != null && !path.getMetadata().isRoot() ? ColumnMetadata.getColumnMetadata(path).getJdbcType() : null;
```

Comment:
The test case is ok, but I'd fix this differently, since root path usage is only one case where this breaks. Maybe
```
Integer sqlType = null;
if (path) {
    ColumnMetadata metadata = ColumnMetadata.getColumnMetadata(path);
    if (metadata.hasJdbcType()) {
        jdbcType = metadata.getJdbcType();
    }
}
```